clang/test/CodeGenSYCL/uses_aspects.cpp

```diff
@@ -57,10 +57,11 @@ void foo() {
     h.single_task<class kernel_name_1>(f1);
   });
 }
-// CHECK: !intel_types_that_use_aspects = !{![[TYPE1:[0-9]+]], ![[TYPE2:[0-9]+]]}
-// CHECK: [[TYPE1]] = !{!"class.Type1WithAspect", i32 1}
-// CHECK: [[TYPE2]] = !{!"class.Type2WithAspect", i32 5, i32 1}
-// CHECK: [[EMPTYASPECTS]] = !{}
-// CHECK: [[ASPECTS1]] = !{i32 1}
-// CHECK: [[ASPECTS2]] = !{i32 5, i32 2}
-// CHECK: [[ASPECTS3]] = !{i32 0}
+// CHECK-DAG: !intel_types_that_use_aspects = !{![[TYPE1:[0-9]+]], ![[TYPE2:[0-9]+]]}
+// CHECK-DAG: [[TYPE1]] = !{!"class.Type1WithAspect", i32 1}
+// CHECK-DAG: [[TYPE2]] = !{!"class.Type2WithAspect", i32 5, i32 1}
+// CHECK-DAG: [[EMPTYASPECTS]] = !{}
+// CHECK-DAG: [[ASPECTS1]] = !{i32 1}
+// CHECK-DAG: [[ASPECTS2]] = !{i32 5, i32 2}
+// CHECK-DAG: [[ASPECTS3]] = !{i32 0}
+// CHECK-DAG: [[ASPECTS4]] = !{i32 1, i32 5}
```

Comment:
This does not seem correct. Aspect 5 corresponds to fp16. Why is this generated for `func6`

Reply:
See my comment above, `func6` uses `Type2WithAspect`, which is annotated with `fp16`